### PR TITLE
APP-3334 - Adding TextEllipsis

### PR DESCRIPTION
**Pull request by axeleriksson147**
https://perzoinc.atlassian.net/browse/APP-3334

- Added TextEllipsis component
-- Will show Tooltip when text is ellipsed
-- You can set number of rows before ellipse

Screenshot 2021-07-26 at 16 13 40

Two row ellipse
Screenshot 2021-07-26 at 21 33 47

## Todo
https://github.com/SymphonyPlatformSolutions/symphony-bdk-ui-toolkit-styles/pull/120 - Needs to merge first.

## Discussion
Should tooltip on ellipsed and on hover be optional? Can add another flag for it.

### CODE REVIEW

**Review by symphony-adnane — COMMENTED** [review]
LGTM pending on styles changes to merge

**Comment by symphony-adnane** [comment]
@axeleriksson147 regarding your question. 
I think it makes sense to have the tooltip on hover enabled by default, but we should have a way to disable it 
So a flag enableTooltip default to true, could be set to false, is welcome!
Thanks again very interesting addition ;)

File: src/components/text-ellipsis/TextEllipsis.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+TextEllipsis.defaultProps = {
+  rows: 1,
+  tooltipPlacement: 'top',
+}
+
+const isTextTruncated = (element: EventTarget & Element) => {
+  const { scrollWidth, scrollHeight, clientWidth, clientHeight} = element;
+
+  if(scrollHeight > clientHeight) {
```
Comment (swasunb) [inline src/components/text-ellipsis/TextEllipsis.tsx:83]:
```suggestion
  return (scrollHeight > clientHeight) || (scrollWidth > clientWidth)
```

File: src/components/text-ellipsis/TextEllipsis.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import Tooltip from '../tooltip';
+
+interface TextEllipsisProps extends Omit<React.HTMLProps<HTMLDivElement>, 'type'> {
```
[the remark this answers is not in the document]
Comment (swasunb) [reply src/components/text-ellipsis/TextEllipsis.tsx:5]: ah i saw otherProps that it receives. So it will.

**Review by swasunb — APPROVED** [review]

**Review by symphony-adnane — APPROVED** [review]

**Pull request merged**